app/views/candidates/registrations/subject_preferences/_form.html.erb

```diff
@@ -0,0 +1,55 @@
+<h1 class="govuk-heading-l">We need some more details</h1>
+<p>The following will be used to help schools offer you school experience placements.</p>
+
+<%= form_for subject_preference, url: candidates_school_registrations_subject_preference_path do |f| %>
+  <%= GovukElementsErrorsHelper.error_summary subject_preference, 'There is a problem', '' %>
+
+  <%= f.radio_button_fieldset :degree_stage do |fieldset| %>
+    <% Candidates::Registrations::SubjectPreference.degree_stages.each do |degree_stage| %>
+      <% if degree_stage == Candidates::Registrations::SubjectPreference::DEGREE_STAGE_REQUIRING_EXPLANATION %>
+        <%= fieldset.radio_input degree_stage do %>
+          <%= fieldset.text_area :degree_stage_explaination %>
+        <% end %>
+      <% else %>
+        <%= fieldset.radio_input degree_stage %>
+      <% end %>
+    <% end %>
+  <% end %>
+
+  <div class="govuk-form-group">
+    <%= f.label :degree_subject, class: 'govuk-label' %>
+    <%= f.select :degree_subject,
+      options_for_select(Candidates::Registrations::SubjectPreference.degree_subjects, f.object.degree_subject),
+      { prompt: '', },
+      { class: 'govuk-select' } %>
+  </div>
+
+  <%= f.radio_button_fieldset :teaching_stage do |fieldset| %>
+    <% Candidates::Registrations::SubjectPreference.teaching_stages.each do |teaching_stage| %>
+      <%= fieldset.radio_input teaching_stage %>
+    <% end %>
+  <% end %>
+
+  <fieldset class="govuk-fieldset">
+    <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
+      <h1 class="govuk-fieldset__heading">What subject do you want to teach?</h1>
+    </legend>
+    <div class="govuk-form-group">
+      <%= f.label :subject_first_choice, class: 'govuk-label' %>
+      <%= f.select :subject_first_choice,
```

Comment:
Would this be simpler as a `collection_select` - you can use `:to_s` as the method for key and value I believe if using it with a plain Array